src/main/java/de/tum/in/www1/artemis/service/QuizService.java

```diff
@@ -0,0 +1,295 @@
+            dndQuestion.setQuizQuestionStatistic(quizQuestionStatistic);
+            quizQuestionStatistic.setQuizQuestion(dndQuestion);
+        }
+
+        for (var dropLocation : dndQuestion.getDropLocations()) {
+            quizQuestionStatistic.addDropLocation(dropLocation);
+        }
+
+        // if a dropLocation was removed then remove the associated AnswerCounters implicitly
+        Set<DropLocationCounter> dropLocationCounterToDelete = new HashSet<>();
+        findTobeDeletedComponent(quizQuestionStatistic.getDropLocationCounters(), dropLocationCounter -> {
+            if (!(dndQuestion.getDropLocations().contains(dropLocationCounter.getDropLocation()))) {
+                dropLocationCounter.setDropLocation(null);
+                dropLocationCounterToDelete.add(dropLocationCounter);
+            }
+            return null;
+        });
+        quizQuestionStatistic.getDropLocationCounters().removeAll(dropLocationCounterToDelete);
+
+        // save references as index to prevent Hibernate Persistence problem
+        saveCorrectMappingsInIndices(dndQuestion);
+    }
+
+    private void fixReferenceShortAnswer(ShortAnswerQuestion saQuestion) {
+        var quizQuestionStatistic = (ShortAnswerQuestionStatistic) saQuestion.getQuizQuestionStatistic();
+        if (quizQuestionStatistic == null) {
+            quizQuestionStatistic = new ShortAnswerQuestionStatistic();
+            saQuestion.setQuizQuestionStatistic(quizQuestionStatistic);
+            quizQuestionStatistic.setQuizQuestion(saQuestion);
+        }
+
+        for (var spot : saQuestion.getSpots()) {
+            spot.setQuestion(saQuestion);
+            quizQuestionStatistic.addSpot(spot);
+        }
+
+        // if a spot was removed then remove the associated spotCounters implicitly
+        Set<ShortAnswerSpotCounter> spotCounterToDelete = new HashSet<>();
+        findTobeDeletedComponent(quizQuestionStatistic.getShortAnswerSpotCounters(), spotCounter -> {
+            if (!(saQuestion.getSpots().contains(spotCounter.getSpot()))) {
+                spotCounter.setSpot(null);
+                spotCounterToDelete.add(spotCounter);
+            }
+            return null;
+        });
+        quizQuestionStatistic.getShortAnswerSpotCounters().removeAll(spotCounterToDelete);
+
+        // save references as index to prevent Hibernate Persistence problem
+        saveCorrectMappingsInIndicesShortAnswer(saQuestion);
+    }
+
+    /**
+     * Call the given deleteCallback for each given component where id is not null
+     *
+     * @param components     the components for which the given deleteCallback to be applied to
+     * @param deleteCallback the callback to be called for each component
+     */
+    private <T1 extends QuizQuestionStatisticComponent<T2, T3, T4>, T2 extends QuizQuestionStatistic, T3 extends QuizQuestionComponent<T4>, T4 extends QuizQuestion> void findTobeDeletedComponent(
```

Comment:
I would also suggest resolving the enumeration here and below

Reply:
enumeration is removed